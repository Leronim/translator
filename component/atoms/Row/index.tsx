import React from 'react';
import { View, StyleSheet } from 'react-native';

interface Props {
    children: React.ReactNode,
    style?: {
        marginTop: number | undefined
    }
}

const Row: React.FC<Props> = (props: Props) => {
    const combineStyles = StyleSheet.flatten([props.style, styles.row])
    
    return (
        <View style={combineStyles}>
            {props.children}
        </View>
    )
}

const styles = StyleSheet.create({
    row: {
        width: "100%",
        paddingRight: 20, 
        paddingLeft: 20,
        flex: 1
    }
})

export default Row;