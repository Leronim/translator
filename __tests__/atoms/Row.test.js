import React from 'react';
import { EventEmitter, Text } from 'react-native';
import Row from '../../component/atoms/Row';
import { render } from '@testing-library/react-native';

test('Row render', () => {
    const tree = render(<Row/>).toJSON();
    expect(tree).toMatchSnapshot();
})

test('Row render children', () => {
    const messageChildren = "Hello";

    const { UNSAFE_getAllByType } = render(
        <Row>
            <Text>{messageChildren}</Text>
        </Row>
    )
    const [ text ] = UNSAFE_getAllByType(Text);

    expect(text.props.children).toEqual(messageChildren);
})