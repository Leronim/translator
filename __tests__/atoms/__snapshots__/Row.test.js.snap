// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row render 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingLeft": 20,
      "paddingRight": 20,
      "width": "100%",
    }
  }
/>
`;
